lib/samson/form_builder.rb

```diff
@@ -50,8 +52,12 @@ def actions(delete: false, &block)
 
     # html regexp input ... allows blank values, does not know case insensitive so use a-zA-Z, does not know \A\z
     # FYI: could also read from validations: _validators[attribute].first.options[:with]
+    # js patterns need to match the full inout "/foo" will not match "/foobar" ... so we enforce \A + \z
     def translate_regex_to_js(pattern)
-      pattern.source.sub('\\A', '^').sub('\\z', '$') if pattern
+      pattern = pattern.source
+      raise ArgumentError, "pattern needs \\A" unless pattern.sub!('\\A', '^')
```

Comment:
that rails method does a bunch of strange things ... `gsub(/\s/ , '')` etc ...
it means that it enforces \A and \z because patterns will not match when they do not match the whole string ... so enforcing \A\z to make users use complete patterns